src/security/security-tests/src/test/java/org/geoserver/security/impl/DefaultResourceAccessManagerAuthTest.java

```diff
@@ -38,6 +38,12 @@ public void testWideOpen() throws Exception {
         checkUserAccessFlat(manager, anonymous, true, true);
     }
 
+    @Test
+    public void testAdminProperties() throws Exception {
+        ResourceAccessManager manager = buildAccessManager("admin.properties");
+        checkUserAccessFlat(manager, root, true, true);
```

Comment:
this method is testing that the canAccess methods return true according to the rules you set. I guess this was happening  even before your fix. The test should also test the getAccessLimits for the workspace doesn't return null but the WorkspaceLimits instance.